File: implementation/scipamato/scipamato-common-persistence-api/src/main/java/ch/difty/scipamato/common/config/ApplicationProperties.java
package ch.difty.scipamato.common.config;

import java.io.Serializable;

/**
 * Provides access to property based configuration parameters.
 *
 * @author u.joss
 */
public interface ApplicationProperties extends Serializable {

    /**
     * @return the build version of the application.
     */
    String getBuildVersion();

    /**
     * @return the localization the application starts with.
     */
    String getDefaultLocalization();

    /**
     * @return the brand name
     */
    String getBrand();

    /**
     * @return the pageTitle if present - otherwise brand name
     */
    String getTitleOrBrand();

    /**
     * @return The base url in PubMed to query by pm_id.
     */
    String getPubmedBaseUrl();

    /**
     * @return The base url in PubMed to query by pm_id.
     */
    Integer getRedirectFromPort();

}

File: common/common-persistence-api/src/main/kotlin/ch/difty/scipamato/common/config/ApplicationProperties.kt
package ch.difty.scipamato.common.config

import java.io.Serializable

/**
 * Provides access to property based configuration parameters.
 *
 * @author u.joss
 */
interface ApplicationProperties : Serializable {
    /**
     * @return the build version of the application.
     */
    val buildVersion: String?

    /**
     * @return the localization the application starts with.
     */
    val defaultLocalization: String

    /**
     * @return the brand name
     */
    val brand: String

    /**
     * @return the pageTitle if present - otherwise brand name
     */
    val titleOrBrand: String

    /**
     * @return The base url in PubMed to query by pm_id.
     */
    val pubmedBaseUrl: String

    /**
     * @return The port from where an unsecured http connection is forward to the secured port.
     */
    val redirectFromPort: Int?

    /**
     * @return the threshold above which a multiselectbox
     * may activate the action box to select all/none entries
     */
    val multiSelectBoxActionBoxWithMoreEntriesThan: Int

    /**
     * @return the url pointing to the CMS page with the study search page
     */
    val cmsUrlSearchPage: String?
}
